### DISCUSSION TITLE
DAG run finished notification from REST API

### DISCUSSION THREAD

**Discussion by johanneszellinger** [apache/airflow#22039] [Q&A] [answered]
I have a web-server, which triggers DAGs via the REST API. I want to know when a DAG is finished in order to collect the results. As far as I can tell I have two options:

- Periodically poll the DAG status via the REST API
- Have another DAG with a script that notifies my web-server, which is automatically triggered at the end of each DAG run. 

Are these the correct approaches, or is there a built-in system for this kind of use-case?

**Comment by josh-fell** [comment, marked as answer]
A couple options come to mind depending on how you are communicating/what your webserver is expecting:

1. You could use [callback](https://airflow.apache.org/docs/apache-airflow/2.2.1/logging-monitoring/callbacks.html) functionality within Airflow for this. Write a function that communicates with your webserver and set that callable as the `on_success_callback` of the last task in your DAG.
2. If you want to have the explicit visibility in your DAG of communicating back to the webserver, create a new task -- maybe the  [`SimpleHttpOperator`](https://github.com/apache/airflow/blob/main/airflow/providers/http/operators/http.py) to send a POST request, `PythonOperator`, etc. -- to interact with the webserver however you need to.
